src/core/logger.ts

```diff
@@ -1,24 +1,46 @@
 import { stringify } from './utils/collections';
 import Pusher from './pusher';
 
+function globalLog(message) {
+  if (global.console && global.console.log) {
+    global.console.log(message);
+  }
+}
+
+function globalLogWarn(message) {
+  if (global.console && global.console.warn) {
+    global.console.warn(message);
+  } else {
+    globalLog(message);
+  }
+}
+
+function globalLogError(message) {
+  if (global.console && global.console.error) {
+    global.console.error(message);
+  } else {
+    globalLogWarn(message);
+  }
+}
+
+function log(defaultLoggingFunction, ...args: any[]) {
```

Comment:
Not sure I understand the name `defaultLoggingFunction`. This argument also doesn't have a type signature.

Reply:
`defaultLoggingFunction` is the log function to use if the user of the library has not specified anything. Can you think of a better name? If not perhaps a parameter doc would help.